website/src/pages/StatusPage/StatusPage.tsx

```diff
@@ -32,23 +44,95 @@ export default function StatusPage() {
     `${import.meta.env.VITE_API_URL}recent`
   );
 
+  const [filters, setFilters] = useState({
+    type: '',
+    status: '',
+    source: '',
+  });
+
+  const filterData = (data: dataTypes[]) => {
+    return data.filter((item) => {
+      const itemType = item.type_of ? item.type_of.toString() : '';
+      const itemSource = item.source ? item.source.toString() : '';
+      const itemStatus = item.status ? item.status.toString() : '';
+
+      const matchesType =
+        filters.type === '' || filters.type === undefined || itemType === filters.type;
+      const matchesSource =
+        filters.source === '' || filters.source === undefined || itemSource === filters.source;
+      const matchesStatus =
+        filters.status === '' || filters.status === undefined || itemStatus === filters.status;
+
+      return (matchesType || filters.type === '' || filters.type === undefined) &&
+      (matchesSource || filters.source === '' || filters.source === undefined) &&
+      (matchesStatus || filters.status === '' || filters.status === undefined);
+    });
+  };
+
+  const handleFilterChange = (e: React.ChangeEvent<HTMLSelectElement>) => {
+    const { name, value } = e.target;
+    setFilters((prevFilters) => ({ ...prevFilters, [name]: value }));
+  };
+
+  const filteredDataQueue = filterData(dataQueue) as dataTypes[]; 
+  const filteredPreviousDataExe = filterData(dataPreviousExe) as dataTypes[];
+
   return (
     <>
       <Hero />
 
       <div className="border-accent border mt-5" />
 
+      {/* FILTERS OPTIONS*/}
+      <div>
+        <label>
```

Comment:
This are working fine 🚀 As you mentioned offline, we should change the style a bit though, using a select like this (https://ui.shadcn.com/docs/components/select) in a card centered above the first table or even right below the text in the Hero.

Reply:
done  👍